### ISSUE TITLE
cumulative sum/prod dtypes inconsistent with integer inputs

### ISSUE THREAD

**Issue by jackd** [keras-team/keras#18730]
`ops.cumsum` and `ops.cumprod` call backend implementations that return tensors with inconsistent dtypes when the input array is an integer dtype.

Additionally, the no-axis `compute_output_spec` implementation for both `Operation`s returns `int32`, regardless of the input dtype.

Backends all have a `dtype` argument which can be used, but when not provided, the following behaviour is observed:

- numpy: promote signed ints to `int64`, unsigned ints to `uint64`
- torch / tensorflow: promote to `int64`
- jax: same as input array (though this is inconsistent with documentation - see [issue](https://github.com/google/jax/issues/18399)

Happy to put in a PR, just need to know what policy to implement.

**Comment by SuryanarayanaY** [comment]
Hi @jackd ,

I hope you are referring to [keras.ops.cumsum](https://github.com/keras-team/keras/blob/master/keras/ops/numpy.py#L1825) and [keras.ops.cumprod](https://github.com/keras-team/keras/blob/master/keras/ops/numpy.py#L1790) respectively.

I have replicated the reported issue and attached gists here for [tf](https://colab.sandbox.google.com/gist/SuryanarayanaY/ac1301950143f6f083a82fcb628052a6/18730_tf_backend.ipynb) and [torch](https://colab.sandbox.google.com/gist/SuryanarayanaY/ed3edda568e44efaac924b2956b5679e/18730_torch_backend.ipynb) as well.

These Ops are calling numpy backend where in all integers are being converted into int64 if dtype is not provided explicitly for `numpy.cumsum` and `numpy.cumprod`

**Comment by jackd** [comment]
Resolved in #18813
